### ISSUE TITLE
Preserve fixed version when running unleash:perform-tycho

### ISSUE THREAD

**Issue by kessltho** [shillner/unleash-maven-plugin#72]
I have an eclipse rcp project with the following plugins:
```
myProject (version 1.13.0-SNAPSHOT)
\-- plugins
    |-- pluginA (version 1.13.0-SNAPSHOT  inherited from 'myProject')
    \-- pluginB (fixed version 7.0.0)`
```

- "myProject" specifies a version "1.13.0-SNAPSHOT" and two maven modules "pluginA" and "pluginB" in its POM file.
- "pluginA" has no explicit version defined, it inherits its version from its parent (myProject).
- "pluginB" also has "myProject" as parent but specifies a fixed version 7.0.0.

I would like to create a release of my project with the version 1.13.0 but the version of "pluginB" should be preserved.

When I run unleash:perform-tycho in my Jenkins Unleash Release Build I see the following output:
```
[INFO] Making changes in /app/devenv/jenkins/data/workspace/MYPROJECT/unleash-release-job/plugins/pluginA
[INFO]   pom.xml//project/version: 1.13.0-SNAPSHOT => 1.13.0
[INFO]   pom.xml//project/parent/version: 1.13.0-SNAPSHOT => 1.13.0
[INFO]   META-INF/MANIFEST.MF//Bundle-Version: 1.13.0.qualifier => 1.13.0
[INFO] Making changes in /app/devenv/jenkins/data/workspace/MYPROJECT/unleash-release-job/plugins/pluginB
[INFO]   pom.xml//project/parent/version: 1.13.0-SNAPSHOT => 1.13.0
[INFO]   pom.xml//project/version: 7.0.0 => 1.13.0
[INFO]   META-INF/MANIFEST.MF//Bundle-Version: 7.0.0 => 1.13.0
```

How can I tell Unleash NOT to modify the version of "pluginB" during the release build (or override the version for "pluginB" with a specific version)?

**Comment by shillner** [comment]
Hi, I'm currently not quite sure about how this is possible since the version upgrading is performed using the tycho-versions mechanism. I'll have a look on this and come back to you soon. Actually I'd expect this to be possible but the tycho mechanisms behave a bit strange sometimes.

**Comment by shillner** [comment]
Hi, I've tried to reproduce your issue but without success. I used a reduced setup of a colleague which looks like the following:

parent (v 0.1.0-SNAPSHOT)
\-- products
    +-- target (v 0.1.0-SNAPSHOT  inherited from parent)
\-- src/core
    +-- model (v 1)

Calling `mvn unleash:perform-tycho -X -B` produces the following:

```
[INFO] 21:33:18,398  Calculating required versions for all modules.
[DEBUG] 21:33:18,398    Versions of module com.itemis.unleash.test:com.itemis.unleash.test.parent:
[DEBUG] 21:33:18,398            PRE_RELEASE = 0.1.0-SNAPSHOT
[DEBUG] 21:33:18,399            RELEASE = 0.1.0
[DEBUG] 21:33:18,399            POST_RELEASE = 0.1.1-SNAPSHOT
[DEBUG] 21:33:18,399    Versions of module com.itemis.unleash.test:com.itemis.unleash.test.targetdefinition:
[DEBUG] 21:33:18,399            PRE_RELEASE = 0.1.0-SNAPSHOT
[DEBUG] 21:33:18,399            RELEASE = 0.1.0
[DEBUG] 21:33:18,399            POST_RELEASE = 0.1.1-SNAPSHOT
[DEBUG] 21:33:18,399    Versions of module com.itemis.unleash.test:com.itemis.unleash.test.core.model:
[DEBUG] 21:33:18,399            PRE_RELEASE = 1
[DEBUG] 21:33:18,399            RELEASE = 1
[DEBUG] 21:33:18,399            POST_RELEASE = 2-SNAPSHOT
[INFO] 21:33:18,399  Updating project modules with release versions (POM and MANIFEST versions)
[INFO] Making changes in C:\Users\Stanley\unleash\test
[INFO]   pom.xml//project/version: 0.1.0-SNAPSHOT => 0.1.0
[INFO] Making changes in C:\Users\Stanley\unleash\test\products\target
[INFO]   pom.xml//project/version: 0.1.0-SNAPSHOT => 0.1.0
[INFO]   pom.xml//project/parent/version: 0.1.0-SNAPSHOT => 0.1.0
[INFO] Making changes in C:\Users\Stanley\unleash\test\src\core\com.itemis.unleash.test.core.model
[INFO]   pom.xml//project/parent/version: 0.1.0-SNAPSHOT => 0.1.0
```

For the tycho versioning engine it is important that the versions of the pom and the manifest are in sync to work correctly!

**Comment by kessltho** [comment]
I created a new mini-project with the same setup and I was able to reproduce the problem.
I found out that the problem only occurs if the development/releaseVersions are specified when running "perform-tycho". If the versions are omitted and unleash calculates the versions by its default version upgrade strategy everything works as expected.

parent (v 1.13.0-SNAPSHOT)
-- products
+-- target (v 1.13.0-SNAPSHOT inherited from parent)
-- src/core
+-- business (v 7.0.0)
+-- model (v 1.13.0-SNAPSHOT inherited from parent)

`mvn unleash:perform-tycho -X -B ` produces 
```
[INFO] Making changes in C:\Daten\dev\source\unleash-tycho\trunk\unleash-tycho
[INFO]   pom.xml//project/version: 1.13.0-SNAPSHOT => 1.13.0
[INFO] Making changes in C:\Daten\dev\source\unleash-tycho\trunk\unleash-tycho\src\core\business
[INFO]   pom.xml//project/parent/version: 1.13.0-SNAPSHOT => 1.13.0
[INFO] Making changes in C:\Daten\dev\source\unleash-tycho\trunk\unleash-tycho\src\core\model
[INFO]   pom.xml//project/version: 1.13.0-SNAPSHOT => 1.13.0
[INFO]   pom.xml//project/parent/version: 1.13.0-SNAPSHOT => 1.13.0
```

`mvn unleash:perform-tycho -X -B -Dunleash.releaseVersion=1.13.0 -Dunleash.developmentVersion=1.14.0-SNAPSHOT` produces 
```
[INFO] Making changes in C:\Daten\dev\source\unleash-tycho\trunk\unleash-tycho
[INFO]   pom.xml//project/version: 1.13.0-SNAPSHOT => 1.13.0
[INFO] Making changes in C:\Daten\dev\source\unleash-tycho\trunk\unleash-tycho\src\core\business
[INFO]   pom.xml//project/parent/version: 1.13.0-SNAPSHOT => 1.13.0
[INFO]   pom.xml//project/version: 7.0.0 => 1.13.0
[INFO] Making changes in C:\Daten\dev\source\unleash-tycho\trunk\unleash-tycho\src\core\model
[INFO]   pom.xml//project/version: 1.13.0-SNAPSHOT => 1.13.0
[INFO]   pom.xml//project/parent/version: 1.13.0-SNAPSHOT => 1.13.0
```
Version 7.0.0 from module "business" is changed although it is not a Snapshot version.

**Comment by shillner** [comment]
Yes sure, you are right and this is the intended behavior. As the [Wiki documentation](https://github.com/shillner/unleash-maven-plugin/wiki/unleash%3Aperform-tycho#releaseversion) says, this version will be assigned to all modules!

Preserving fixed versions of already released modules would require another parameter. I'll consider this for the next release but this will likely take a while.

**Comment by kessltho** [comment]
I was not sure whether this was the intended behaviour. An additional parameter for that would be very helpful. Until then I will continue to use my workaround (prepare the desired versions in a pre-step and then release using the default version upgrade strategy instead of specifying a release version.
Thank you for your help and the quick response times!

**Comment by shillner** [comment]
Will be available in 2.6.0! Please find the documentation in the project Wiki once the release is available.
